File: gdb/hello.c

/*
 *
 */
#include <stdio.h>

void test(char *msg)
{
        printf("%s",msg);
}

void print(char* msg)
{
        printf("%s\n", "hello print");
        test(msg);
}

int main()
{
        print("This is Hello World2\n");
        return 0;
}

File: test/hello/hello.py

import sys

def hello2():
	return "Hello World2\n"

def hello():
	return "Hello World\n"

def main(name):
	func = getattr(sys.modules[__name__], name)
	return func()

if __name__ == "__main__":
	main(sys.argv[1])
